ui/lib/apps/Statement/pages/List/TimeRangeSelector.tsx

```diff
@@ -124,6 +124,44 @@ export function calcValidStatementTimeRange(
   }
 }
 
+function calcCommonTimeRange(
+  minServerDataTime: number,
+  maxServerDataTime: number
+): { enabled: boolean; value: number }[] {
+  if (!maxServerDataTime) {
+    return RECENT_SECONDS.map((s) => ({ enabled: false, value: s }))
+  }
+
+  const calcInterval =
+    maxServerDataTime +
+    getLocalTimeInterval(maxServerDataTime) -
+    minServerDataTime
+
+  return RECENT_SECONDS.map((s) => ({ enabled: s <= calcInterval, value: s }))
+}
+
+// use time interval to calculate valid time range when dropdown change visible state
+//
+// ----first time open drop down----------...---------every time open drop down----
+//                                (time interval)
+// -----------Date.now()------------------...------------------Date.now()----------
+let getLocalTimeInterval = (time: number): number => {
+  let firstTime: number
+  if (!!time) {
+    firstTime = calcAbsIntervalFromNow(time)
+  }
+
+  getLocalTimeInterval = (time: number): number => {
```

Comment:
This usage makes me a little confused, change the function definition inside itself?

Reply:
Oh, this is a buggy impl. Let me re-impl it.